test/KernelWrap/kernel_wrap.cl

```diff
@@ -0,0 +1,24 @@
+
+__attribute__((work_group_size_hint(1,1,1)))
+__attribute__((reqd_work_group_size(1,1,1)))
+__kernel void add(global int *A, global int *B) {
+  // Add 1 to each element of the input buffer.
+  for (unsigned int i = 0; i < get_global_size(0); i++) {
+    B[i] = A[i] + 1;
+  }
+}
+
+// kernel1.clspv
+kernel void main_kernel(global int* A, global int* B) {
+  // Call the function.
+  add(A, B);
+}
+
+// CHECK-DAG: [[kernel_name:%[a-zA-Z0-9_]+]] = OpString "add"
```

Comment:
you are not using `kernel_name` after. If it is not useful, don't store it. But in your case, I agree you need to get it, but you also need to check where it is used.
Same for the argName (make sure to use different names for each function. It will be easier to maintain.